tests/appsec/test_remoteconfiguration.py

```diff
@@ -30,36 +44,166 @@ def test_rc_enabled_by_default(tracer):
 
 
 @pytest.mark.parametrize(
-    "appsec_enabled, rc_value, expected",
+    "appsec_enabled, rc_value",
     [
-        ("", None, False),
-        ("", None, False),
-        ("", False, False),
-        ("", True, True),
-        ("false", None, False),
-        ("true", None, True),
-        ("false", False, False),
-        ("false", True, False),
-        ("true", False, False),
-        ("true", True, True),
+        ("", True),
+        ("true", True),
     ],
 )
-def test_rc_activation_states(tracer, appsec_enabled, rc_value, expected):
+def test_rc_activation_states_on(tracer, appsec_enabled, rc_value, remote_config_worker):
+    tracer.configure(appsec_enabled=False)
     with override_env({APPSEC_ENV: appsec_enabled}):
         if appsec_enabled == "":
             del os.environ[APPSEC_ENV]
 
         appsec_rc_reload_features(tracer)(None, {"asm": {"enabled": rc_value}})
         result = _set_and_get_appsec_tags(tracer)
-        if expected:
-            assert "triggers" in result
-        else:
-            assert result is None
+        assert "triggers" in result
+
+
+@pytest.mark.parametrize(
+    "appsec_enabled, rc_value",
+    [
+        ("", False),
+        ("false", False),
+        ("false", True),
+        ("true", False),
+    ],
+)
+def test_rc_activation_states_off(tracer, appsec_enabled, rc_value, remote_config_worker):
+    tracer.configure(appsec_enabled=True)
+    with override_env({APPSEC_ENV: appsec_enabled}):
+        if appsec_enabled == "":
+            del os.environ[APPSEC_ENV]
+
+        rc_config = {"asm": {"enabled": True}}
+        if rc_value is False:
+            rc_config = False
+
+        appsec_rc_reload_features(tracer)(None, rc_config)
+        result = _set_and_get_appsec_tags(tracer)
+        assert result is None
+
+
+@pytest.mark.parametrize(
+    "rc_enabled, capability",
+    [
+        ("true", "Bg=="),
+        ("false", ""),
+    ],
+)
+def test_rc_capabilities(rc_enabled, capability):
+    with override_env({"DD_REMOTE_CONFIGURATION_ENABLED": rc_enabled}):
+        assert _appsec_rc_capabilities() == capability
+
+
+@mock.patch.object(RemoteConfig, "_check_remote_config_enable_in_agent")
+def test_rc_activation_validate_products(mock_check_remote_config_enable_in_agent, tracer, remote_config_worker):
+    mock_check_remote_config_enable_in_agent.return_value = True
+    tracer.configure(appsec_enabled=False, api_version="v0.4")
+
+    rc_config = {"asm": {"enabled": True}}
+
+    assert not RemoteConfig._worker
+
+    appsec_rc_reload_features(tracer)(None, rc_config)
+
+    assert RemoteConfig._worker._client._products["ASM_DATA"]
+
+
+def test_rc_activation_ip_blocking_data(tracer, remote_config_worker):
+    with override_env({APPSEC_ENV: "true"}):
+        tracer.configure(appsec_enabled=True, api_version="v0.4")
+        rc_config = {
+            "rules_data": [
+                {
+                    "data": [{"expiration": 1755346879, "value": "user8"}],
```

Comment:
future date?

Reply:
I wonder if we should just do `int(time.time()) + 20000)` or something similar to what you did below.

Reply:
No, this date isn't needed, is part or `data_with_expiration` object. We use in this test `ip_with_expiration` but we want to check the rules with two or more objects to be sure the function hasn't unexpected overrides